### PR TITLE
GEOS-7557: vectortiles - remove degenerate geometries produced by clipping

### PR DESCRIPTION
**Pull request by DBlasby**
https://osgeo-org.atlassian.net/browse/GEOS-7557
https://github.com/geoserver/geoserver/wiki/GSIP-138

After clipping, the resulting geometries are filtered so that lower-dimensional results are removed.

I also re-formatted the code so it follows GS coding standards.

### CODE REVIEW

File: src/community/vectortiles/src/main/java/org/geoserver/wms/vector/PipelineBuilder.java [hunk 1]
```diff
@@ -302,11 +313,123 @@ protected Geometry _run(Geometry geom) throws Exception {
...
+        private Geometry collectionClip(GeometryCollection geom) throws Exception {
+            ArrayList<Geometry> result = new ArrayList<Geometry>();
+            for (int t = 0; t < geom.getNumGeometries(); t++) {
+                Geometry g = geom.getGeometryN(0);
+                Geometry clipped = _run(g); // gets the non-degenerative of the result
+                if ((clipped != null) && (!clipped.isEmpty()))
+                    result.add(clipped);
+            }
+            if (result.size() == 0)
+                return null;
```
Comment (tbarsballe) [inline src/community/vectortiles/src/main/java/org/geoserver/wms/vector/PipelineBuilder.java]: Doesn't follow coding conventions, needs `{ ... }`

File: src/community/vectortiles/src/main/java/org/geoserver/wms/vector/PipelineBuilder.java [hunk 2]
```diff
@@ -302,11 +313,123 @@ protected Geometry _run(Geometry geom) throws Exception {
...
+        }
+
+        private Geometry onlyPolygon(Geometry result) {
+            if ((result instanceof Polygon) || (result instanceof MultiPolygon))
+                return result;
+            List polys = com.vividsolutions.jts.geom.util.PolygonExtracter.getPolygons(result);
+            if (polys.size() == 0)
+                return null;
+            if (polys.size() == 1)
+                return (Polygon) polys.get(0);
```
Comment (tbarsballe) [inline src/community/vectortiles/src/main/java/org/geoserver/wms/vector/PipelineBuilder.java]: More missing braces

File: src/community/vectortiles/src/main/java/org/geoserver/wms/vector/PipelineBuilder.java [hunk 3]
```diff
@@ -302,11 +313,123 @@ protected Geometry _run(Geometry geom) throws Exception {
...
+        }
+
+        private Geometry onlyLines(Geometry result) {
+            if ((result instanceof LineString) || (result instanceof MultiLineString))
+                return result;
+            List lines = com.vividsolutions.jts.geom.util.LineStringExtracter.getLines(result);
+            if (lines.size() == 0)
+                return null;
+            if (lines.size() == 1)
+                return (LineString) lines.get(0);
```
Comment (tbarsballe) [inline src/community/vectortiles/src/main/java/org/geoserver/wms/vector/PipelineBuilder.java]: More missing braces

**Comment by jodygarnett** [comment]
@groldan can you review this please?

**Comment by aaime** [comment]
Merge conflicts... we're losing it! :-)

**Comment by DBlasby** [comment]
Closed - this is superceeded by https://github.com/geoserver/geoserver/pull/1887

**Pull request closed without merging**